common-lib/common/signal/log.libsonnet

```diff
@@ -0,0 +1,68 @@
+local g = import '../g.libsonnet';
+local utils = import '../utils.libsonnet';
+local base = import './base.libsonnet';
+local signalUtils = import './utils.libsonnet';
+local lokiQuery = g.query.loki;
+base {
+  new(
+    name,
+    type,
+    unit,
+    nameShort,
+    description,
+    aggLevel,
+    aggFunction,
+    vars,
+    datasource,
+    sourceMaps,
+  ):
+    base.new(
+      name,
+      type,
+      unit,
+      nameShort,
+      description,
+      aggLevel,
+      aggFunction,
+      vars,
+      datasource,
+      sourceMaps=sourceMaps,
+    )
+    {
+      common(type)::
+        // override panel-wide --mixed-- datasource
+        lokiQuery.withDatasource('${%s}' % datasource)
+        + g.panel.timeSeries.panelOptions.withDescription(description),
+
+
+      //Return as grafana panel target(query+legend)
+      asTarget(name=name):
+        lokiQuery.new(
+          '${%s}' % datasource,
+          self.asPanelExpression(),
+        )
+        + lokiQuery.withRefId(name),
+
+      asGauge()::
+        error 'asGauge() is not supported for log metrics.',
+      asStat()::
+        error 'asStat() is not supported for log metrics.',
+      asTimeSeries()::
+        error 'asTimeSeries() is not supported for log metrics.',
```

Comment:
I know you're not using `asTimeSeries()` above on line 35, but it is still pulling the `timeSeries.panelOptions.withDescription` method, so I wonder if we want to pull generics like `withDescription` out on its own?

Reply:
thanks, changed to logs for cleaner looks.